Fix how and when default properties are used

ligi opened the pull request:
the main reason behind this PR was that I got this exception when merging the latest changes:
```
➜  ethereumj-studio git:(master) gradle clean run
Parallel execution is an incubating feature.
[buildinfo] Properties file path was not found! (Relevant only for builds running on a CI Server)
:ethereumj-studio:clean
:ethereumj-core:antlr4
:ethereumj-studio:processResources UP-TO-DATE
:ethereumj-core:compileJava UP-TO-DATE
:ethereumj-core:processResources UP-TO-DATE
:ethereumj-core:classes UP-TO-DATE
:ethereumj-core:jar UP-TO-DATE
:ethereumj-studio:compileJava
warning: Supported source version 'RELEASE_6' from annotation processor 'org.antlr.v4.runtime.misc.NullUsageProcessor' less than -source '1.8'
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
1 warning
:ethereumj-studio:classes
:ethereumj-studio:run
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
log4j:ERROR Could not instantiate class [org.apache.log4j.rolling.RollingFileAppender].
java.lang.ClassNotFoundException: org.apache.log4j.rolling.RollingFileAppender
	at java.net.URLClassLoader$1.run(URLClassLoader.java:372)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:361)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:360)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:308)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:263)
	at org.apache.log4j.helpers.Loader.loadClass(Loader.java:198)
	at org.apache.log4j.helpers.OptionConverter.instantiateByClassName(OptionConverter.java:327)
	at org.apache.log4j.helpers.OptionConverter.instantiateByKey(OptionConverter.java:124)
	at org.apache.log4j.PropertyConfigurator.parseAppender(PropertyConfigurator.java:785)
	at org.apache.log4j.PropertyConfigurator.parseCategory(PropertyConfigurator.java:768)
	at org.apache.log4j.PropertyConfigurator.configureRootCategory(PropertyConfigurator.java:648)
	at org.apache.log4j.PropertyConfigurator.doConfigure(PropertyConfigurator.java:514)
	at org.apache.log4j.PropertyConfigurator.doConfigure(PropertyConfigurator.java:580)
	at org.apache.log4j.helpers.OptionConverter.selectAndConfigure(OptionConverter.java:526)
	at org.apache.log4j.LogManager.<clinit>(LogManager.java:127)
	at org.apache.log4j.Logger.getLogger(Logger.java:104)
	at org.apache.commons.logging.impl.Log4JLogger.getLogger(Log4JLogger.java:289)
	at org.apache.commons.logging.impl.Log4JLogger.<init>(Log4JLogger.java:109)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:422)
	at org.apache.commons.logging.impl.LogFactoryImpl.createLogFromClass(LogFactoryImpl.java:1116)
	at org.apache.commons.logging.impl.LogFactoryImpl.discoverLogImplementation(LogFactoryImpl.java:914)
	at org.apache.commons.logging.impl.LogFactoryImpl.newInstance(LogFactoryImpl.java:604)
	at org.apache.commons.logging.impl.LogFactoryImpl.getInstance(LogFactoryImpl.java:336)
	at org.apache.commons.logging.impl.LogFactoryImpl.getInstance(LogFactoryImpl.java:310)
	at org.apache.commons.logging.LogFactory.getLog(LogFactory.java:685)
	at org.springframework.context.support.AbstractApplicationContext.<init>(AbstractApplicationContext.java:154)
	at org.springframework.context.support.GenericApplicationContext.<init>(GenericApplicationContext.java:100)
	at org.springframework.context.annotation.AnnotationConfigApplicationContext.<init>(AnnotationConfigApplicationContext.java:60)
	at org.springframework.context.annotation.AnnotationConfigApplicationContext.<init>(AnnotationConfigApplicationContext.java:82)
	at org.ethereum.facade.EthereumFactory.createEthereum(EthereumFactory.java:110)
	at org.ethereum.gui.UIEthereumManager.<clinit>(UIEthereumManager.java:11)
	at org.ethereum.gui.ToolBar.<init>(ToolBar.java:235)
	at org.ethereum.gui.ToolBar.lambda$main$13(ToolBar.java:240)
	at org.ethereum.gui.ToolBar$$Lambda$1/424058530.run(Unknown Source)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:749)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:702)
	at java.awt.EventQueue$3.run(EventQueue.java:696)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:75)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:719)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
log4j:ERROR Could not instantiate appender named "file".
Exception in thread "AWT-EventQueue-0" java.lang.ExceptionInInitializerError
	at org.ethereum.gui.ToolBar.<init>(ToolBar.java:235)
	at org.ethereum.gui.ToolBar.lambda$main$13(ToolBar.java:240)
	at org.ethereum.gui.ToolBar$$Lambda$1/424058530.run(Unknown Source)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:749)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:702)
	at java.awt.EventQueue$3.run(EventQueue.java:696)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:75)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:719)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'ethereumFactory': Injection of autowired dependencies failed; nested exception is org.springframework.beans.factory.BeanCreationException: Could not autowire field: org.ethereum.facade.Ethereum org.ethereum.facade.EthereumFactory.eth; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'ethereumImpl': Injection of autowired dependencies failed; nested exception is org.springframework.beans.factory.BeanCreationException: Could not autowire field: org.ethereum.manager.WorldManager org.ethereum.facade.EthereumImpl.worldManager; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'worldManager': Injection of autowired dependencies failed; nested exception is org.springframework.beans.factory.BeanCreationException: Could not autowire field: private org.ethereum.facade.Blockchain org.ethereum.manager.WorldManager.blockchain; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'blockchainImpl': Injection of autowired dependencies failed; nested exception is org.springframework.beans.factory.BeanCreationException: Could not autowire field: private org.ethereum.facade.Repository org.ethereum.core.BlockchainImpl.repository; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'repositoryImpl' defined in URL [jar:file:/home/ligi/git/3rd/ethereumj/ethereumj-core/build/libs/ethereumj-core-0.8.1-SNAPSHOT.jar!/org/ethereum/db/RepositoryImpl.class]: Instantiation of bean failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.ethereum.db.RepositoryImpl]: Constructor threw exception; nested exception is java.lang.NullPointerException
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.postProcessPropertyValues(AutowiredAnnotanPostProcessor.java:334)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.populateBean(AbstractAutowireCapableBeanFactory.java:1204)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:538)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:476)
	at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:302)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:229)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:298)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:193)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:762)
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:757)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:480)
	at org.springframework.context.annotation.AnnotationConfigApplicationContext.<init>(AnnotationConfigApplicationContext.java:84)
	at org.ethereum.facade.EthereumFactory.createEthereum(EthereumFactory.java:110)
	at org.ethereum.gui.UIEthereumManager.<clinit>(UIEthereumManager.java:11)
	... 17 more
Caused by: org.springframework.beans.factory.BeanCreationException: Could not autowire field: org.ethereum.facade.Ethereum org.ethereum.facade.EthereumFactory.eth; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'ethereumImpl': Injection of autowired dependencies failed; nested exception is org.springframework.beans.factory.BeanCreationException: Could not autowire field: org.ethereum.manager.WorldManager org.ethereum.                                                                              facade.EthereumImpl.worldManager; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'worldManager': Injection of autowired dependencies failed; nested exception is org.springframework.beans.factory.BeanCreationException: Could not autowire field: private org.ethereum.facade.Blockchain org.ethereum.manager.WorldManager.blockchain; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'blockchainImpl': Injection of autowired dependencies failed; nested exception is org.springframework.beans.factory.BeanCreationException: Could not autowire field: private org.ethereum.facade.Repository org.ethereum.core.BlockchainImpl.repository; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'repositoryImpl' defined in URL [jar:file:/home/ligi/git/3rd/ethereumj/ethereumj-core/build/libs/ethereumj-core-0.8.1-SNAPSHOT.jar!/org/ethereum/db/RepositoryImpl.class]: Instantiation of bean failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.ethereum.db.RepositoryImpl]: Constructor threw exception; nested exception is java.lang.NullPointerException
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:555)
	at org.springframework.beans.factory.annotation.InjectionMetadata.inject(InjectionMetadata.java:87)
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.postProcessPropertyValues(AutowiredAnnotationBeanPostProcessor.java:331)
	... 30 more
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'ethereumImpl': Injection of autowired dependencies failed; nested exception is org.springframework.beans.factory.BeanCreationException: Could not autowire field: org.ethereum.manager.WorldManager org.ethereum.facade.EthereumImpl.worldManager; nested e     xception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'worldManager': Injection of autowired dependencies failed; nested exception is org.springframework.beans.factory.BeanCreationException: Could not autowire field: private org.ethereum.facade.Blockchain org.ethereum.manager.WorldManager.blockchain; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'blockchainImpl': Injection of autowired dependencies failed; nested exception is org.springframework.beans.factory.BeanCreationException: Could not autowire field: private org.ethereum.facade.Repository org.ethereum.core.BlockchainImpl.repository; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'repositoryImpl' defined in URL [jar:file:/home/ligi/git/3rd/ethereumj/ethereumj-core/build/libs/ethereumj-core-0.8.1-SNAPSHOT.jar!/org/ethereum/db/RepositoryImpl.class]: Instantiation of bean failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.ethereum.db.RepositoryImpl]: Constructor threw exception; nested exception is java.lang.NullPointerException
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.postProcessPropertyValues(AutowiredAnnotationBeanPostProcessor.java:334)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.populateBean(AbstractAutowireCapableBeanFactory.java:1204)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:538)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:476)
	at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:302)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:229)
	at org.spmework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:298)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:193)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.findAutowireCandidates(DefaultListableBeanFactory.java:1127)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1051)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:949)
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:527)
	... 32 more
Caused by: org.springframework.beans.factory.BeanCreationException: Could not autowire field: org.ethereum.manager.WorldManager org.ethereum.facade.EthereumImpl.worldManager; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'worldManager': Injection of autowired dependencies failed; nested exception is org.springframework.beans.factory.BeanCreationException: Could not autowire field: private org.ethereum.facade.Blockchain org.ethereum.manager.WorldManager.blockchain; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'blockchainImpl': Injection of autowired dependencies failed; nested exception is org.springframework.beans.factory.BeanCreationException: Could not autowire field: private org.ethereum.facade.Repository org.ethereum.core.BlockchainImpl.repository; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'repositoryImpl' defined in URL [jar:file:/home/ligi/git/3rd/ethereumj/ethereumj-core/build/libs/ethereumj-core-0.8.1-SNAPSHOT.jar!/org/ethereum/db/RepositoryImpl.class]: Instantiation of bean failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantia                                                                                          te [org.ethereum.db.RepositoryImpl]: Constructor threw exception; nested exception is java.lang.NullPointerException
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:555)
	at org.springframework.beans.factory.annotation.InjectionMetadata.inject(InjectionMetadata.java:87)
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.postProcessPropertyValues(AutowiredAnnotationBeanPostProcessor.java:331)
	... 43 more
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'worldManager': Injection of autowired dependencies failed; nested exception is org.springframework.beans.factory.BeanCreationException: Could not autowire field: private org.ethereum.facade.Blockchain org.ethereum.manager.WorldManager.blockchain; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'blockchainImpl': Injection of autowired dependencies failed; nested exception is org.springframework.beans.factory.BeanCreationException: Could not autowire field: private org.ethereum.facade.Repository org.ethereum.core.BlockchainImpl.repository; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'repositoryImpl' defined in URL [jar:file:/home/ligi/git/3rd/ethereumj/ethereumj-core/build/libs/ethereumj-core-0.8.1-SNAPSHOT.jar!/org/ethereum/db/RepositoryImpl.class]: Instantiation of bean failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.ethereum.db.RepositoryImpl]: Constructor threw exception; nested exception is java.lang.NullPointerException
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.postProcessPropertyValues(AutowiredAnnotationBeanPostProcessor.java:334)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.populateBean(AbstractAutowireBeanFactory.java:1204)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:538)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:476)
	at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:302)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:229)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:298)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:193)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.findAutowireCandidates(DefaultListableBeanFactory.java:1127)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1051)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:949)
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:527)
	... 45 more
Caused by: org.springframework.beans.factory.BeanCreationException: Could not autowire field: private org.ethereum.facade.Blockchain org.ethereum.manager.WorldManager.blockchain; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'blockchainImpl': Injection of autowired dependencies failed; nested exception is org.springframework.beans.factory.BeanCreationException: Could not autowire field: private org.ethereum.facade.Repository org.ethereum.core.BlockchainImpl.repository; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'repositoryImpl' defined in URL [jar:file:/home/ligi/git/3rd/ethereumj/ethereumj-core/build/li           bs/ethereumj-core-0.8.1-SNAPSHOT.jar!/org/ethereum/db/RepositoryImpl.class]: Instantiation of bean failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.ethereum.db.RepositoryImpl]: Constructor threw exception; nested exception is java.lang.NullPointerException
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:555)
	at org.springframework.beans.factory.annotation.InjectionMetadata.inject(InjectionMetadata.java:87)
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.postProcessPropertyValues(AutowiredAnnotationBeanPostProcessor.java:331)
	... 56 more
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'blockchainImpl': Injection of autowired dependencies failed; nested exception is org.springframework.beans.factory.BeanCreationException: Could not autowire field: private org.ethereum.facade.Repository org.ethereum.core.BlockchainImpl.repository; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'repositoryImpl' defined in URL [jar:file:/home/ligi/git/3rd/ethereumj/ethereumj-core/build/libs/ethereumj-core-0.8.1-SNAPSHOT.jar!/org/ethereum/db/RepositoryImpl.class]: Instantiation of bean failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.ethereum.db.RepositoryImpl]: Constructor threw exception; nested exception is java.lang.NullPointerException
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.postProcessPropertyValues(AutowiredAnnotationBeanPostProcessor.java:334)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.populateBean(AbstractAutowireCapableBeanFactory.java:1204)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFacta:538)v
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:476)
	at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:302)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:229)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:298)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:193)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.findAutowireCandidates(DefaultListableBeanFactory.java:1127)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1051)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:949)
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:527)
	... 58 more
Caused by: org.springframework.beans.factory.BeanCreationException: Could not autowire field: private org.ethereum.facade.Repository org.ethereum.core.BlockchainImpl.repository; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'repositoryImpl' defined in URL [jar:file:/home/ligi/git/3rd/ethereumj/ethereumj-core/build/libs/ethereumj-core-0.8.1-SNAPSHOT.jar!/org/ethereum/db/RepositoryImpl.class]: Instantiation of bean failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.ethereum.db.RepositoryImpl]: Constructor threw exception; nested exception is java.lang.NullPointerException
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:555)
	at org.springframework.betory.annotation.InjectionMetadata.inject(InjectionMetadata.java:87)
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.postProcessPropertyValues(AutowiredAnnotationBeanPostProcessor.java:331)
	... 69 more
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'repositoryImpl' defined in URL [jar:file:/home/ligi/git/3rd/ethereumj/ethereumj-core/build/libs/ethereumj-core-0.8.1-SNAPSHOT.jar!/org/ethereum/db/RepositoryImpl.class]: Instantiation of bean failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.ethereum.db.RepositoryImpl]: Constructor threw exception; nested exception is java.lang.NullPointerException
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateBean(AbstractAutowireCapableBeanFactory.java:1095)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1040)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:505)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:476)
	at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:302)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:229)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:298)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:193)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.findAutowireCandidates(DefaultListableBeanFactory.java:1127)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1051)
	at org.springfrabeans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:949)
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:527)
	... 71 more
Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.ethereum.db.RepositoryImpl]: Constructor threw exception; nested exception is java.lang.NullPointerException
	at org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:163)
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:89)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateBean(AbstractAutowireCapableBeanFactory.java:1088)
	... 82 more
Caused by: java.lang.NullPointerException
	at org.ethereum.db.DatabaseImpl.<init>(DatabaseImpl.java:35)
	at org.ethereum.db.RepositoryImpl.<init>(RepositoryImpl.java:60)
	at org.ethereum.db.RepositoryImpl.<init>(RepositoryImpl.java:56)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:422)
	at org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:147)
	... 84 more

BUILD SUCCESSFUL

Total time: 8.11 secs
```

investigation showed that it was rooted in when and how defaults are used - also while I was here I did some other minor code improvements in this area

cbeams commented on ethereumj-core/src/main/java/org/ethereum/config/SystemProperties.java:
```diff
@@ -262,30 +234,25 @@ public String rootHashStart() {
     public List<String> peerCapabilities() {
-        if (prop.isEmpty()) return DEFAULT_PROTOCOL_LIST;
-        String capabilitiesList = prop.getProperty("peer.capabilities");
+        final String capabilitiesList = prop.getProperty("peer.capabilities", DEFAULT_PROTOCOL_LIST);
```
no need for a `final` local var in this context.

cbeams commented:
Changes look good in general, save for the compilation error. Thanks, @ligi.
Also, please [capitalize the first word of commit messages](http://chris.beams.io/posts/git-commit/#capitalize) from here out. Cheers.

ligi commented:
thanks @cbeams  for reviewing! updated the PR - changed the commit-msg - the final thing is a habit from work - we make everything final there which can be final - so you know instantly when reading that something will not be changed - but I removed it from here now.
